lib/auth.js

```diff
@@ -52,6 +52,35 @@ var auth = {
     } else {
       return cb(null, false);
     }
+  },
+  /**
+   * New endpoint required as autentication ($fh.auth) requests not handled
+   * correctly in local development/Studio Preview -
+   * the js sdk dertermines envrironment mode from url string.
+   * New endpoint determines environment mode from env var,
+   * optionally sends request (localAuth - for local dev)
+   * and calls user provided function (cb).
+   * @param  {object} req
+   * @param  {function} localAuth - optional for local development.
+   * @param  {object} cb
+   */
+  performAuth: function (req, localAuth, cb) {
+    // "FH_USE_LOCAL_DB" env var determines if the app is running in local mode.
+    if (process.env.FH_USE_LOCAL_DB) {
+      // In local mode, invoke user function/object
+      if (_.isFunction(localAuth)) {
+        return localAuth(req, cb);
+      }
+      else if (_.isObject(localAuth)) {
+        return cb(undefined, localAuth);
+      } 
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)